src/game/collide.c

```diff
@@ -864,6 +864,30 @@ void AlignLaraPosition(PHD_VECTOR *vec, ITEM_INFO *item, ITEM_INFO *lara_item)
     phd_PopMatrix();
 }
 
+int32_t MoveLaraPositionAnim(
+    PHD_VECTOR *vec, ITEM_INFO *item, ITEM_INFO *lara_item)
+{
+    PHD_3DPOS dest;
+    dest.x_rot = item->pos.x_rot;
+    dest.y_rot = item->pos.y_rot;
+    dest.z_rot = item->pos.z_rot;
+    phd_PushUnitMatrix();
+    phd_RotYXZ(item->pos.y_rot, item->pos.x_rot, item->pos.z_rot);
+    PHD_MATRIX *mptr = g_PhdMatrixPtr;
+    dest.x = item->pos.x
+        + ((vec->x * mptr->_00 + vec->y * mptr->_01 + vec->z * mptr->_02)
+           >> W2V_SHIFT);
+    dest.y = item->pos.y
+        + ((vec->x * mptr->_10 + vec->y * mptr->_11 + vec->z * mptr->_12)
+           >> W2V_SHIFT);
+    dest.z = item->pos.z
+        + ((vec->x * mptr->_20 + vec->y * mptr->_21 + vec->z * mptr->_22)
+           >> W2V_SHIFT);
+    phd_PopMatrix();
+    return Move3DPosTo3DPosAnim(
+        &lara_item->pos, &dest, 12, MOVE_ANG, lara_item);
```

Comment:
`MOVE_VELOCITY`?

Reply:
Yes it seems TR5 used a slower value than TR1 (which was only used for underwater movement). `MOVE_SPEED` in TR1 is 16 while this uses 12.

Also I meant to link the functions from TR5 for reference. Some of the values are still magic numbers. I was hoping you might have some name suggestions because I wasn't sure.

https://github.com/Trxyebeep/TOMB5/blob/73e8a036a76290c14641f957768af4ec45fa4459/TOMB5/game/collide.cpp#L886

Reply:
I don't see any remaining magic numbers apart from `>> 16` which I believe is related to high precision multiplications but I might be mistaken and as such I don't have suggestions. Did I miss something else?